### PR TITLE
feat: add MsgSupplyCollateral

### PR DESCRIPTION
**Pull request by toteki**
This might or might not be okay to merge in a minor version (due to adding a message type). Supports a common frontend use case.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by toteki** [comment]
This PR should be evaluated for inclusion in `v3.1.0`

**Review by robert-zaremba — COMMENTED** [review]
I don't think we need it. API, especially on proto level, should not have redundancy when possible.
The same effect can be achieved with one transaction with multiple messages.

File: x/leverage/keeper/msg_server.go [hunk 1]
```diff
@@ -108,6 +108,51 @@ func (s msgServer) Collateralize(
...
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	supplierAddr, err := sdk.AccAddressFromBech32(msg.Supplier)
+	if err != nil {
+		return nil, err
+	}
+	uToken, err := s.keeper.Supply(ctx, supplierAddr, msg.Asset)
+	if err != nil {
+		return nil, err
+	}
```
Comment (robert-zaremba) [inline x/leverage/keeper/msg_server.go:124]: how about calling `s.Supply` and `s.Collaterize` directly here (by creating the appropriate message)?

**Comment by robert-zaremba** [comment]
> this could have unintended consequences if we feed them into MsgServer (I'm not 100% sure).

You don't go through MsgServer, and no need to check signatures. You directly call the handler.
In fact even if you would go through MsgServer (which is not clearly supported today), it won't check signatures nor call ValidateBasic (which is happening at different stage)

**Review by robert-zaremba — APPROVED** [review]
pre-approving. See https://github.com/umee-network/umee/pull/1479#issuecomment-1270837149 . That would be my preference, but I'm not against using the current way. I just prefer to avoid possible duplications.

**Pull request merged**